ddtrace/internal/writer.py

```diff
@@ -64,19 +67,41 @@ def flush_queue(self):
             log.error('error while filtering traces: {0}'.format(err))
             return
 
-        if not traces:
-            return
+        traces_flushed = len(traces)
+        traces_flushed_error = 0
 
         # If we have data, let's try to send it.
         traces_responses = self.api.send_traces(traces)
         for response in traces_responses:
             if isinstance(response, Exception) or response.status >= 400:
                 self._log_error_status(response)
+                traces_flushed_error += 1
             elif self._priority_sampler:
                 result_traces_json = response.get_json()
                 if result_traces_json and 'rate_by_service' in result_traces_json:
                     self._priority_sampler.set_sample_rate_by_service(result_traces_json['rate_by_service'])
 
+        # Dump statistics
+        # NOTE: Do not use the buffering of dogstatsd as it's not thread-safe
+        # https://github.com/DataDog/datadogpy/issues/439
+        if self.dogstatsd:
+            self.dogstatsd.increment('datadog.tracer.traces.api.requests', len(traces_responses))
+            for status, grouped_responses in itertools.groupby(
+                    sorted(traces_responses, key=lambda r: r.status),
+                    key=lambda r: r.status):
+                self.dogstatsd.increment('datadog.tracer.traces.api.responses.count',
+                                         len(list(grouped_responses)),
+                                         tags=['status:%d' % status])
+            if hasattr(time, 'thread_time_ns'):
+                self.dogstatsd.increment('datadog.tracer.writer.cpu_time', time.thread_time_ns())
+            self.dogstatsd.increment('datadog.tracer.traces.flushed', traces_flushed)
+            self.dogstatsd.increment('datadog.tracer.traces.flushed.errors', traces_flushed_error)
+            dropped, enqueued, enqueued_lengths = self._trace_queue.reset_stats()
+            self.dogstatsd.increment('datadog.tracer.traces.dropped', dropped)
+            self.dogstatsd.increment('datadog.tracer.traces.enqueued', enqueued)
+            self.dogstatsd.increment('datadog.tracer.traces.enqueued_lengths', enqueued_lengths)
+            self.dogstatsd.gauge('datadog.tracer.traces.queue_max_length', self._trace_queue.maxsize)
```

Comment:
This seems like it will be static at 1k always, any reason to have this?

Reply:
I think it's great to have this being sent in the case is ever become more dynamic in the future. You can draw it on your chart dynamically and see how "full" your queue is at any point without having to dig into the source code to know the Q size.